### DISCUSSION TITLE
Welcome to memex Discussions!

### DISCUSSION THREAD

**Discussion by kormyen** [kormyen/memex#19] [General]
## 👋 Welcome!
Testing Discussions as a place to:
  * Ask questions about memex.
  * Share ideas.
  * Engage with other memex users.
  * Build together 💪.

**Comment by domnantas** [comment]
hi!
you haven't updated memex in a while. have you stopped using it?
i'm in early stages of experimenting with building my own memex/exocortex/digital garden, along with minimal cms, inspired by webring people (including you!).
`home` hasn't been updated in a while either. why is that? i don't mean to guilt trip you into writing, i'm genuinely curious as i have a feeling i'm going to enjoy building the tool more than actually using it myself. has this been the case for you too?

**Reply by kormyen** [reply]
Hey Domnantas!
I have stopped using Memex lately yes. 
Still interested and have plans to return to both projects.
I have been really busy with life stuff for the last while, but slowly chipping away with tutorials on different UI options for mobile. I'm most interested in departing the kind of local webapp thing and moving to more of a simple server for both. I personally want a more instant and accessible way to add and edit data with them.

**Reply by domnantas** [reply]
i see. we share a similar issue – when i was shaping the idea of my own digital garden, i realized i want to edit the data with as little friction as possible. that means editing on mobile and maybe even offline. i'd also like to make it as simple as possible and avoid databases.

i imagine a minimal github (or other git hosts) web client under `/admin` route where a particular repo content is listed, you can add media and edit the files with a simple text editor and your changes would be commited. similar to https://github.com/netlify/netlify-cms, [forestry.io](https://www.forestry.io/) or [logseq](https://logseq.com/), but entirely self-hosted. barebones text editor would not limit you to markdown/json only, and you could make changes to .ndtl, .tbtl or even the .c or .js files to change your site generation logic. something like that would make it possible to do edits from anywhere, simple automation could then trigger builds and serve the static files.

i'm struggling to come up with a good authentication solution. all i want is a single button in `/admin` page which would open github authentication page. after granting the permission, the token would be saved to cookies/localStorage and then i'd be able to make commits to github from the tiny cms. the problem is, this flow requires some sort of backend middleware _i think_, i'm not sure if it's safe to do all of that client-side, store `client_id` in the public repo. i'll need to explore netllify-cms to figure out how it works. another option would be to use netlify identity services (but i'd prefer to not rely on 3rd parties), or self-host https://github.com/netlify/git-gateway. all of this is pretty confusing to me, i don't want to risk exposing my account by doing something stupid.

as for offline functionality, it could utilize pwa, service workers. that would make it "installable" as a mobile app and able to store edits locally until connection is restored and they are published in the background. at least that's how i imagine it, not sure if service workers could be used that way. i'd like to avoid js as much as possible, but for functionality like that it may be worth the sacrifice.

also, i'm not sure how to make it "integratable" to other projects, for example, the memex. i think it would be useful to have `https://kormyen.github.io/memex/admin` route where you authorize with github and have a simple, mobile friendly interface to edit the contents.
